rfc/0009-rancher-integration-charts.md

```diff
@@ -0,0 +1,187 @@
+- Submit list of upstream images to be mirrored to
+  https://github.com/rancher/image-mirror. For normal container images, this is
+  no problem. For Wasm modules, they will not be supported by the Rancher
+  DockerHub repository.
+- Provide our own `serviceaccount`.
+- Define k8s min/max version with `kubeVersion` in `Chart.yaml`.
+- Add scheduling options at a pod (affinity) and node level
+  (taints/tolerations/selectors).
+- Provide `annotations.catalog.cattle.io/upstream-version`. This could be
+  automatically bumped on release.
+- Provide `questions.yaml` for defaults on Rancher Explorer UI.
+
+The Kubewarden stack has the following dependencies:
+- Designated main chart: `kubewarden-controller`. This will be the main charts
+  to build a Rancher package from.
+- Hard dependencies: 
+  * CRDs: in `kubewarden-crds` chart. On the main Kubewarden chart
+    (`kubewarden-controller`), set `catalog.cattle.io/auto-install:
+    kubewarden-crds=match` that points to the right CRD version.
+    This annotation accepts `match`, or a hardcoded version (e.g: `1.1.0`)
+    We ought to set it to a hardcoded version, to be able to release patch
+    versions.  of `kubewarden-controller` independently.
+  * `cert-manager`: considered a precondition of Rancher Explorer. Rancher
+    instructs users to install it prior to installing Rancher, so we can assume
+    it is always present.
+- Soft (optional) dependencies:
+  * `grafana`, `kube-prometheus-stack`: provided by `rancher-monitoring` chart.
+  * `jaeger-operator`: provided by `rancher-tracing` chart.
+  * `open-telemetry/opentelemetry-operator`: not yet in Rancher's repository. We
+    will need to submit it (possibly to
+    https://github.com/rancher/partner-charts) and maintain.
+  * `kubewarden-defaults`
+
+  For the ones depending on CRDs (`kubewarden-defaults`,
+    `opentelemetry-operator`, `jaeger-operator`), Rancher charts follow a 3 step
+  process (this process is only for optional dependencies, see previous hard
+  dependencies annotations):
+  * Wrap the custom resource in a conditional that is tied to a values.yaml
+    parameter.
+  * Talk to the Rancher Explorer UI team to implement a solution and create a
+  dashboard issue. Make them aware that if a certain CRD is defined we want to
+  flip the new added value to true. If there is configuration available on that
+  resource then it would be grayed out in the UI if the CRD is not available,
+  and editable if the CRD is available.
+  * Document this in our charts.
+  
+  For the ones not depending on CRDs (`grafana`, `prometheus`), we expect end
+  users to deploy/reuse a chart for them and configure the endpoints themselves.
+
+All of the modifications to our codebases will be carried in `rancher-X`
+branches, where `X` is the tag being targeted.
+They will be feature branches with parent being the commit tagged `X`. Using
+these branches allows for easy rebases and cherry-picks, and documenting
+needed changes per commit. 
```

Comment:
I don't understand why we need separate branches. Are you talking about helm-charts? or also the other repos such as kubewarden-controller or policy-server. Why cannot we just use tags?

Reply:
well, it's a feature branch, if they get merged because it's backwards compatible, all is fine. If things can't get merged, it's on the branch. Also, it's an effort to provide a vendor separation, in case other actors come and want to add modifications needed for their packaging in Kubewarden upstream.